subsys/bluetooth/host/adv.c

```diff
@@ -1139,6 +1139,7 @@ static int le_ext_adv_param_set(struct bt_le_ext_adv *adv,
 	err = bt_id_set_adv_own_addr(adv, param->options, dir_adv,
 				     &cp->own_addr_type);
 	if (err) {
+		net_buf_unref(buf);
```

Comment:
No, it is released by `bt_hci_cmd_send_sync`:
https://github.com/zephyrproject-rtos/zephyr/blob/25e90e7bb0ef04ce087fbba2ddc08e918334cfa6/subsys/bluetooth/host/hci_core.c#L458

Reply:
Right :) 

It can actually return an error without unref'ing the buffer in https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/hci_core.c#L378-L384 but I guess the ASSERT there indicates that it shouldn't happen :D